pin-project-internal/src/pin_project/derive.rs

```diff
@@ -136,6 +136,7 @@ fn proj_allowed_lints(cx: &Context<'_>) -> (TokenStream, TokenStream, TokenStrea
         #proj_mut_allowed_lints
         #[allow(dead_code)] // This lint warns unused fields/variants.
         #[allow(clippy::mut_mut)] // This lint warns `&mut &mut <ty>`.
+        #[allow(clippy::missing_docs_in_private_items)]
```

Comment:
I could add some code that only inserts this in the case of an enum?

We can't fold it into `large_enum_variant` because it's also required in structs in `proj_own`, so that would duplicate the lint.